File: TypeScript/Arrays/rotateArrays.ts
function rotateArray(array: number[], k: number): number[] {
    if (array.length === 0) {
        return [];
    }

    k = k % array.length;

    const temp = array.slice(-k);

    for (let i = array.length - k - 1; i >= 0; i--) {
        array[i + k] = array[i];
    }

    for (let i = 0; i < temp.length; i++) {
        array[i] = temp[i];
    }

    return array;
}

File: Golang/Arrays/rotateArrays.go
package main

func rotateArray(array []int, k int) []int {
	if len(array) == 0 {
		return []int{}
	}

	k = k % len(array)

	temp := make([]int, k)
	copy(temp, array[len(array)-k:])

	for i := len(array) - k - 1; i >= 0; i-- {
		array[i+k] = array[i]
	}

	copy(array, temp)

	return array
}
